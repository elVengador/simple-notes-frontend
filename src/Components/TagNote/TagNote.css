.tag-note {
  padding: 1em;
}
.tag-note__title {
  padding: 0.7em 10px;
  margin: 0em;
  text-align: left;
  background: var(--primary);
  color: var(--text-secondary);
  box-shadow: 3px 3px 5px #9b9b9b;
}

/*# sourceMappingURL=TagNote.css.map */
