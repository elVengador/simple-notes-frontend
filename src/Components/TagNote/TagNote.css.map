{"version":3,"sourceRoot":"","sources":["TagNote.scss"],"names":[],"mappings":"AAAA;EACE;;AACA;EACE;EACA;EAEA;EAEA;EACA;EACA","file":"TagNote.css"}