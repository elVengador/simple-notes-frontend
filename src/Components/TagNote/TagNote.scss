.tag-note {
  padding: 1em;
  &__title {
    padding: 0.7em 10px;
    margin: 0em;

    text-align: left;

    background: var(--primary);
    color: var(--text-secondary);
    box-shadow: 3px 3px 5px rgb(155, 155, 155);
  }
}
