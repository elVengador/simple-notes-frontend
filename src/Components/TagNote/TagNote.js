import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
import './TagNote.css';
//import Note from "../Note/Note";

export default function TagNote({ testId, tag, notes }) {
  const iconDown = <FontAwesomeIcon className="icon" icon={faCaretDown} />;
  /*const componentNotes = notes.map((cur, idx) => (
    <Note key={`note-${idx}`} title={cur} />
  ));*/
  return (
    <div data-testid={testId} className="tag-note">
      <h3 className="tag-note__title">
        {iconDown}
        {tag}:
      </h3>
      {/*componentNotes*/}
    </div>
  );
}
