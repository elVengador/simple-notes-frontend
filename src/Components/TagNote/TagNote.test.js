import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TagNote from "./TagNote";

const tag = "tag-note";
const notes = ["firth note", "second note"];
const componentTagNote = (
  <TagNote testId="id-tag-note" tag={tag} notes={notes} />
);

describe("TagNote", () => {
  test("1.render component", () => {
    const { getByTestId } = render(componentTagNote);
    expect(getByTestId("id-tag-note")).toBeInTheDocument;
    expect(getByTestId("id-tag-note").childNodes.length).toBe(notes.length + 1);
  });

  test("matches snapshot", () => {
    const tree = renderer.create(componentTagNote).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
