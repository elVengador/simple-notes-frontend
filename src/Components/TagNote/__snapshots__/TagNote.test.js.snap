// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagNote matches snapshot 1`] = `
<div
  className="tag-note"
  data-testid="id-tag-note"
>
  <h3
    className="tag-note__title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-caret-down fa-w-10 icon"
      data-icon="caret-down"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        fill="currentColor"
        style={{}}
      />
    </svg>
    tag-note
    :
  </h3>
</div>
`;
